orangecontrib/prototypes/widgets/owinteractions.py

```diff
@@ -0,0 +1,242 @@
+		return self.removed_h[attr1, attr2] - self.gains[attr1] - self.gains[attr2]
+
+
+class Heuristic:
+	def __init__(self, weights):
+		self.n = len(weights)
+		self.w = weights.copy()
+		self.a = np.arange(self.n)
+		self.a = self.a[np.argsort(self.w)]
+
+	def generate_states(self):
+		# prioritize two mid ranked attributes over highest first
+		for s in range(1, self.n * (self.n - 1) // 2):
+			for i in range(max(s - self.n + 1, 0), (s + 1) // 2):
+				yield self.a[i], self.a[s - i]
+
+	def get_states(self, initial_state):
+		states = self.generate_states()
+		if initial_state is not None:
+			while next(states) != initial_state:
+				pass
+			return chain([initial_state], states)
+		return states
+
+
+class InteractionRank(Orange.widgets.data.owcorrelations.CorrelationRank):
+	IntRole = next(gui.OrangeUserRole)
+	RemovedRole = next(gui.OrangeUserRole)
+
+	def __init__(self, *args):
+		super().__init__(*args)
+		self.interaction = None
+
+	def initialize(self):
+		super(Orange.widgets.data.owcorrelations.CorrelationRank, self).initialize()
+		data = self.master.disc_data
+		self.attrs = data and data.domain.attributes
+		self.model_proxy.setFilterKeyColumn(-1)
+		self.heuristic = None
+		self.use_heuristic = False
+		self.sel_feature_index = self.master.feature and data.domain.index(self.master.feature)
+		if data:
+			self.interaction = Interaction(data)
+			self.use_heuristic = len(data) * len(self.attrs) ** 2 > SIZE_LIMIT
+			if self.use_heuristic and not self.sel_feature_index:
+				self.heuristic = Heuristic(self.interaction.gains)
+
+	def compute_score(self, state):
+		attr1, attr2 = state
+		h = self.interaction.class_h
+		score = self.interaction(attr1, attr2) / h
+		gain1 = self.interaction.gains[attr1] / h
+		gain2 = self.interaction.gains[attr2] / h
+		removed = self.interaction.removed_h[attr1, attr2] / h
+		return -score, score, gain1, gain2, removed
+
+	def row_for_state(self, score, state):
+		attrs = sorted((self.attrs[x] for x in state), key=attrgetter("name"))
+		attr_items = []
+		for i, attr in enumerate(attrs):
+			item = QStandardItem(attr.name)
+			item.setData(attrs, self._AttrRole)
+			item.setData(Qt.AlignLeft + Qt.AlignTop, Qt.TextAlignmentRole)
+			item.setToolTip("Attribute Info: {:.1f}%".format(100*score[2+i]))
+			attr_items.append(item)
+		interaction_item = QStandardItem("{:+.1f}%".format(100*score[1]))
+		interaction_item.setData(score[1], self.IntRole)
+		interaction_item.setData(score[4], self.RemovedRole)
+		interaction_item.setData(attrs, self._AttrRole)
+		interaction_item.setData(
+			self.NEGATIVE_COLOR if score[1] < 0 else self.POSITIVE_COLOR,
+			gui.TableBarItem.BarColorRole)
+		interaction_item.setToolTip("Entropy removed: {:.1f}%".format(100*score[4]))
+		return [interaction_item] + attr_items
+
+	def check_preconditions(self):
+		return self.master.disc_data is not None
+
+
+class OWInteractions(Orange.widgets.data.owcorrelations.OWCorrelations):
```

Comment:
If you extend a widget, you should set its property `openclass=True`.

Reply:
As far as I know, this is something that needs to changed in the base orange3 repository, not this one. Should I open a pull request there as well?

Reply:
I think we should wait with this one: let's not open the class. Instead, when this widget gets out of the prototype phase, perhaps create a common superclass for both.